tests/vm/tcastnil.nim

```diff
@@ -0,0 +1,35 @@
+discard """
+  nimout: '''nil
+nil
+nil
+nil
+nil
+nil
+'''
+"""
+
+block:
+  static:
+    let a = cast[pointer](nil)
+    echo a.repr
+
+block:
+  static:
+    echo cast[ptr int](nil).repr
+
+block:
+  const str = cast[ptr int](nil)
+  static:
+    echo str.repr
+
+block:
+  static:
+    echo cast[ptr int](nil).repr
+
+block:
+  static:
+    echo cast[RootRef](nil).repr
+
+block:
+  static:
+    echo cast[cstring](nil).repr
```

Comment:
same comment as previously, doAssert is better; testament only checks that nimout (+friends) is a subset of the stdout output, so if someone later adds some test, it may succeed for the wrong reasons (yes, this happened before).
=>
```nim
doAssert cast[cstring](nil).repr == "nil"
```